src/efftester.ml

```diff
@@ -245,16 +280,19 @@ type unify_solution =
 
 exception No_solution
 
+(* FIXME: what does it do? why? *)
 let rec unify_list = function
   | [] -> []
   | (l, r) :: rest ->
     let sub = unify_list rest in
     (match (subst sub l, subst sub r) with
+    (* FIXME: fix pattern matching *)
     | Unit, Unit -> sub
     | Int, Int -> sub
     | Float, Float -> sub
     | Bool, Bool -> sub
     | String, String -> sub
+    | Option a, Option b -> unify_list [ (a, b) ]
```

Comment:
I suspect that this is not quite right, that it should be `unify_list [ (a, b) ] @ sub` -- see the `List` and `Fun` cases below.